src/hooks/loadables/useLoadTxQueue.ts

```diff
@@ -5,14 +5,31 @@ import useSafeInfo from '../useSafeInfo'
 import { Errors, logError } from '@/services/exceptions'
 import { TxEvent, txSubscribe } from '@/services/tx/txEvents'
 
+const PROPOSAL_EVENTS: (TxEvent.PROPOSED | TxEvent.SIGNATURE_PROPOSED)[] = [
+  TxEvent.PROPOSED,
+  TxEvent.SIGNATURE_PROPOSED,
+]
+
 export const useLoadTxQueue = (): AsyncResult<TransactionListPage> => {
   const { safe, safeAddress, safeLoaded } = useSafeInfo()
   const { chainId, txQueuedTag, txHistoryTag } = safe
   const [proposedTxId, setProposedTxId] = useState<string>()
 
   // Listen to newly proposed txns
   useEffect(() => {
-    return txSubscribe(TxEvent.PROPOSED, ({ txId }) => setProposedTxId(txId))
+    const unsubFns = PROPOSAL_EVENTS.map((event) => {
+      return txSubscribe(event, ({ txId, tx }) => {
+        // User may propose tx, switch wallet and propose signature which would not
+        // trigger getTransactionQueue as proposedTxId wouldn't change
+        const uniqId = `${tx.signatures.size}_${txId}`
+
+        setProposedTxId(uniqId)
+      })
+    })
+
+    return () => {
+      unsubFns.forEach((unsub) => unsub())
+    }
```

Comment:
I suggest we remove this whole reload-queue-on-propose. It’s proving to be tricky and backend is supposed to update the queue via a tag. We might have a bug there if it doesn’t.